### PR TITLE
maintainers/teams: add beam team

### PR DESCRIPTION
**Pull request by yurrriq**
###### Motivation for this change

Add a beam team to better coordinate maintenance of BEAM-related packages and modules.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yurrriq** [comment]
So far, I've included maintainers from related packages:
- @ankhers
- @couchemar
- @filalex77 :interrobang: 
- @gleber
- @happysalada
- @havvy
- @sjmackenzie
- @yurrriq

**Comment by SuperSandro2000** [comment]
@yurrriq please fix the merge conflict and if we have a positive reaction from everyone mentioned (thumbs up the first post maybe) we can merge this.

**Comment by yurrriq** [comment]
Sorry about that, I've rebased now and added @Br1ght0ne.

**Comment by yurrriq** [comment]
Should we update [doc/languages-frameworks/beam.section.md](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/beam.section.md) to mention the team?

**Comment by SuperSandro2000** [comment]
> Should we update [doc/languages-frameworks/beam.section.md](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/beam.section.md?rgh-link-date=2021-01-23T20%3A36%3A52Z) to mention the team?

Might be a good idea.

**Comment by yurrriq** [comment]
With some cursory grepping, I wasn't able to find any mention of teams in other docs :confused:

**Comment by SuperSandro2000** [comment]
> With some cursory grepping, I wasn't able to find any mention of teams in other docs 😕

I did not check this but if the others don't do it then it is fine without it.

**Comment by happysalada** [comment]
@yurrriq thnaks a lot for doing this!

Can you add the generic builder from elixir
and the gleam compiler?

When you have a moment

**Comment by SuperSandro2000** [comment]
> it's deprecated as far as I know and there are no existing maintainers

It would be still good to have some maintainers for it as long as it is in nixpkgs.

**Comment by Havvy** [comment]
My name should be removed from anywhere I'm mentioned, but you should definitely **NOT** create a new team of people without asking for their permission.

File: maintainers/team-list.nix [hunk 1]
```diff
@@ -29,6 +29,20 @@ with lib.maintainers; {
...
 
+  beam = {
+    members = [
+      Br1ght0ne
+      ankhers
+      couchemar
+      gleber
+      happysalada
+      havvy
+      sjmackenzie
```
Comment (sjmackenzie) [inline maintainers/team-list.nix:40]: Please remove my account name.

**Comment by SuperSandro2000** [comment]
> My name should be removed from anywhere I'm mentioned, but you should definitely **NOT** create a new team of people without asking for their permission.

We can't purge logs and activity on GitHub. It is pretty normal to collect people who updated packages and where written as maintainer in a team and ask then for their permission to not miss anyone.

**Comment by DianaOlympos** [comment]
If you need one more maintainer, feel free to add me. I have been doing quite some work around the beam set recently

**Comment by happysalada** [comment]
@yurrriq how about proceeding the other way around. Just adding people who have reacted.
so far I can see @Br1ght0ne, yourself, @DianaOlympos and me.
Let's keep it at that for now, people can always add themselves later on.
If we wait for everybody's approval we might struggle to merge this.
I don't think anyone who doesn't respond within a week will mind being taken off the maintainer's list of a package.

**Comment by yurrriq** [comment]
My intention was for this Draft PR to start these discussions regarding desired team members. Part of the reason it's marked as a draft is to ensure we don't make any changes without consent and buy-in from all involved. I figured it was better to have something tangible to discuss.

To be clear for those that have requested to be removed. As this PR exists currently, that would also remove you as maintainers of the specific BEAM packages. Is that the desired effect?

File: maintainers/team-list.nix (hunk 1 above)
Comment (yurrriq) [reply maintainers/team-list.nix]: Thanks for the response. Removed.

**Comment by yurrriq** [comment]
I've force pushed the following updates:
- Change the members list to those who have actively expressed interest
- Set `maintainers = lib.teams.beam.members` on a few more BEAM-related expressions

I notice the `pgsql`, `webdriver`, and `hex` packages currently have @ericbmerritt as the sole maintainer.

1) Should we set `maintainers = lib.teams.beam.members` there too?
2) Would you like to be included in the BEAM team, Eric?
    - Notably, this additionally removes @tazjin from `pkgs/development/interpreters/elixir/generic-builder.nix`; Would you like to be included in the BEAM team?

**Comment by yurrriq** [comment]
> If you need one more maintainer, feel free to add me. I have been doing quite some work around the beam set recently

@DianaOlympos, it looks like we'll need to add you to `maintainers/maintainer-list.nix`. I can do that now for convenience, otherwise feel free to PR against yurrriq/nixpkgs@beam-team.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -2267,6 +2267,12 @@
     githubId = 265220;
     name = "David Leung";
   };
+  DianaOlympos = {
```
Comment (yurrriq) [inline maintainers/maintainer-list.nix:2270]: @DianaOlympos, please confirm or deny this is acceptable.

**Comment by ankhers** [comment]
I am fine with being added to this list with the caveat that I do not currently have a lot of time, at least until this virus clears up a little more in my area.

**Comment by happysalada** [comment]
@yurrriq I would say yes to setting the beam team for pgsql, webdriver and hex.

**Comment by tazjin** [comment]
@yurrriq Thanks for asking, unfortunately I don't have time for this at the moment (and - again unfortunately - I'm not using any BEAM languages for the time being :))

**Comment by yurrriq** [comment]
> @yurrriq I would say yes to setting the beam team for pgsql, webdriver and hex.

Done. As I see it, we're waiting on feedback from Eric Merritt and Thomas Depierre before marking this ready for broader review.

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (DianaOlympos) [reply maintainers/maintainer-list.nix:2270]: Confirm, all good for me

Comment (yurrriq) [reply maintainers/maintainer-list.nix:2270]: thanks!

**Comment by happysalada** [comment]
@yurrriq @SuperSandro2000 
it looks like @ericbmerritt is busy. How about we merge this for now, we can always make a new PR later?
Let me know if you think this is premature.

**Comment by yurrriq** [comment]
Looks like [his last contribution was in 2016](https://github.com/NixOS/nixpkgs/commits?author=ericbmerritt), so that seems fair to me.

**Comment by gleber** [comment]
I do not have much time for BEAM in Nixpkgs lately. I can not promise swift response, so If you need backup people on the list, feel free to include me.

**Comment by happysalada** [comment]
@SuperSandro2000 when you have a moment, please take a look. No problem if you are busy.

**Comment by Mic92** [comment]
is the list complete now?

**Comment by happysalada** [comment]
@Mic92 yes that is my understanding.

**Comment by yurrriq** [comment]
I'd say it's complete for now anyway.

**Comment by Mic92** [comment]
I could not invite @DianaOlympos to https://github.com/orgs/NixOS/teams/beam, because he is not in the nixos organisation yet.

**Comment by DianaOlympos** [comment]
@Mic92 i am in now

**Comment by Mic92** [comment]
fixed.

**Comment by yurrriq** [comment]
Thanks, @Mic92 and @DianaOlympos!

**Pull request merged**